### PR TITLE
feat: improved latency tests

### PR DESCRIPTION
**Pull request by douglaz**
Closes https://github.com/fedimint/fedimint/issues/3686

### CODE REVIEW

**Review by justinmoon — APPROVED** [review]
LGTM

File: devimint/src/main.rs [hunk 1]
```diff
@@ -119,14 +168,29 @@ pub async fn latency_tests(dev_fed: DevFed) -> Result<()> {
...
-              AVG LN RECV TIME: {ln_recv_time:.3}"
+              REISSUE: {reissue_stats}\n\
+              LN SEND: {ln_sends_stats}\n\
+              LN RECV: {ln_receives_stats}"
     );
+    // FIXME: should be smaller
+    assert!(reissue_stats.median < Duration::from_secs(4));
+    assert!(ln_sends_stats.median < Duration::from_secs(6));
+    assert!(ln_receives_stats.median < Duration::from_secs(6));
+    let factor = 3; // FIXME: should be much smaller
```
Comment (elsirion) [inline devimint/src/main.rs:186]: Let's see if #3694 can get us to reasonable multiples again :)

File: devimint/src/main.rs [hunk 2]
```diff
@@ -24,6 +24,47 @@ use tokio::fs;
...
+        write!(f, ", p90: {:.1}s", self.p90.as_secs_f32())?;
+        write!(f, ", max: {:.1}s", self.max.as_secs_f32())?;
+        write!(f, ", sum: {:.1}s", self.sum.as_secs_f32())?;
+        Ok(())
+    }
+}
+
+fn stats_for(mut v: Vec<Duration>) -> Stats {
+    assert!(!v.is_empty());
+    v.sort();
```
Comment (dpc) [inline devimint/src/main.rs:50]: nit: `sort_unstable` is slightly faster ;D . Not sure how big these are.

**Review by dpc — APPROVED** [review]

File: devimint/src/main.rs (hunk 2 above)
Comment (douglaz) [reply devimint/src/main.rs:50]: 20-30 elements

**Pull request merged**